File: cpp/base/src/main/vsdk/toolkit/io/geometry/ReaderPlyResult.cpp
#include "java/util/ArrayList.txx"
#include "vsdk/toolkit/environment/geometry/surface/TriangleMesh.h"
#include "vsdk/toolkit/io/geometry/ReaderPlyResult.h"

ReaderPlyResult::ReaderPlyResult()
    : triangleMesh(0), pointCloud()
{
}

ReaderPlyResult::~ReaderPlyResult()
{
    if ( triangleMesh != 0 ) {
        delete triangleMesh;
        triangleMesh = 0;
    }
}

TriangleMesh* ReaderPlyResult::detachTriangleMesh()
{
    TriangleMesh* out = triangleMesh;
    triangleMesh = 0;
    return out;
}

File: java/base/src/main/vsdk/toolkit/io/geometry/ReaderPlyResult.java
package vsdk.toolkit.io.geometry;

import java.util.ArrayList;

import vsdk.toolkit.common.linealAlgebra.Vector3Dd;
import vsdk.toolkit.environment.geometry.surface.TriangleMesh;

public class ReaderPlyResult {
    public TriangleMesh triangleMesh;
    public ArrayList<Vector3Dd> pointCloud;

    public ReaderPlyResult() {
        triangleMesh = null;
        pointCloud = new ArrayList<Vector3Dd>();
    }
}
